### PR TITLE
Release 0.7 Fix gym_unity Tests

### PR DESCRIPTION
**Pull request by ervteng**
Changed tests to use Mock entirely rather than using MockCommunicator.

### CODE REVIEW

**Comment by vincentpierre** [comment]
Would it be possible to have a separate method that generates the Mock objects ? This way we can reuse them later in other future tests.

**Comment by ervteng** [comment]
Moved the creation of mocks into helper functions.

File: gym-unity/gym_unity/envs/unity_env.py [hunk 1]
```diff
@@ -113,6 +113,7 @@ def reset(self):
             space.
         """
         info = self._env.reset()[self.brain_name]
+        print(self._env.reset())
```
Comment (vincentpierre) [inline gym-unity/gym_unity/envs/unity_env.py]: Is that intended ?

File: gym-unity/gym_unity/envs/unity_env.py [hunk 2]
```diff
@@ -156,6 +157,12 @@ def step(self, action):
                 action = self._flattener.lookup_action(action)
 
         info = self._env.step(action)[self.brain_name]
+        # print("Hi")
```
Comment (vincentpierre) [inline gym-unity/gym_unity/envs/unity_env.py]: Remove dead code

File: gym-unity/gym_unity/envs/unity_env.py [hunk 3]
```diff
@@ -173,8 +180,9 @@ def _single_step(self, info):
             self.visual_obs = self._preprocess_single(info.visual_observations[0][0, :, :, :])
             default_observation = self.visual_obs
         else:
+            print(info.vector_observations)
```
Comment (vincentpierre) [inline gym-unity/gym_unity/envs/unity_env.py]: print statement

File: gym-unity/gym_unity/envs/unity_env.py [hunk 4]
```diff
@@ -173,8 +180,9 @@ def _single_step(self, info):
...
             default_observation = info.vector_observations[0, :]
-
+        print(default_observation)
```
Comment (vincentpierre) [inline gym-unity/gym_unity/envs/unity_env.py]: print

File: gym-unity/tests/test_gym.py [hunk 5]
```diff
@@ -68,3 +58,52 @@ def test_branched_flatten(mock_env):
...
+
+    :Mock mock_env: A mock UnityEnvironment, usually empty.
+    :Mock mock_brain: A mock Brain object that specifies the params of this environment.
+    :Mock mock_braininfo: A mock BrainInfo object that will be returned at each step and reset.
+    """
+    mock_env.return_value.academy_name = 'MockAcademy'
+    mock_env.return_value.brains = {'MockBrain':mock_brain}
+    mock_env.return_value.external_brain_names = ['MockBrain']
+    mock_env.return_value.reset.return_value = {'MockBrain':mock_braininfo}
+    mock_env.return_value.step.return_value = {'MockBrain':mock_braininfo}
```
Comment (vincentpierre) [inline gym-unity/tests/test_gym.py]: Add white space at the end

Comment (ervteng) [reply gym-unity/tests/test_gym.py]: None of our files have this tho

**Pull request merged**